sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/ModelHelper.java

```diff
@@ -204,4 +207,24 @@ public static ShareFileItemProperties transformFileProperty(FileProperty propert
         return new InternalShareFileItemProperties(property.getCreationTime(), property.getLastAccessTime(),
             property.getLastWriteTime(), property.getChangeTime(), property.getLastModified(), property.getEtag());
     }
+
+    public static HandleItem transformHandleItem(com.azure.storage.file.share.implementation.models.HandleItem handleItem) {
+        return new HandleItem()
+            .setHandleId(handleItem.getHandleId())
+            .setPath(handleItem.getPath().getContent())
```

Comment:
We need to check if the path is encoded, and decoded it if it is.  See https://github.com/Azure/azure-sdk-for-net/pull/30841/files#diff-5e6738dde5a5ea5c21a08c5c191fc7e4bba1e8a6f0855fd0559a85bd0e77bf64R221